app/models/secret_storage.rb

```diff
@@ -50,6 +50,43 @@ def self.keys
     end
   end
 
+  class HashicorpVault
+
+    SECRET_BACKEND ='secret/'.freeze
+    # we don't really want other directories in here,
+    # and there may be other chars that we find we don't like
+    ENCODINGS = {"/": "%2F"}
+
+    def self.read(key)
+      result = Vault.logical.read(SECRET_BACKEND  + self.encode(key))
+      result.data[:secret_data]
+    end
+
+    def self.write(key, data)
```

Comment:
no real choice.  you have directories, directories contain keys.  so there has to be some kinda default key, else we are in the business of trying to search the tree and find keys (and there is no recursive searching etc).